dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/internal/ModuleSelectionConfigImpl.java

```diff
@@ -173,16 +243,55 @@ public List<String> findIncludesNotMatchingAnything(Collection<? extends ModuleI
             }
         }
 
-        return includedNotYetSeen.stream().map(ModuleName::toString).collect(toList());
+        List<ModuleName> includedNotYetSeenButSeenUnderOldNaming = new ArrayList<>();
+        if (serverxmlmode && ProductInfo.getBetaEdition()) {
+            for (Iterator<ModuleName> iterator = includedNotYetSeen.iterator(); iterator.hasNext();) {
+                ModuleName moduleName = iterator.next();
+                for (ModuleInfo moduleInfo : moduleInfos) {
+                    if (matches(moduleName, moduleInfo, false)) {
+                        includedNotYetSeenButSeenUnderOldNaming.add(moduleName);
+                        iterator.remove();
+                        break;
+                    }
+                }
+            }
+
+            for (String unmatchedInclude : includedNotYetSeenButSeenUnderOldNaming.stream().map(ModuleName::toString).collect(toList())) {
+                Tr.warning(tc, MessageConstants.OPENAPI_USING_WRONG_NAME_SOURCE_CWWKO1680W, Constants.MERGE_INCLUDE_CONFIG, unmatchedInclude);
+            }
+        }
+
+        for (String unmatchedInclude : includedNotYetSeen.stream().map(ModuleName::toString).collect(toList())) {
+            Tr.warning(tc, MessageConstants.OPENAPI_MERGE_UNUSED_INCLUDE_CWWKO1667W, Constants.MERGE_INCLUDE_CONFIG, unmatchedInclude);
+        }
+
     }
 
+    /**
+     *
+     * @param name a name we're configured to include in the openAPI documentation
+     * @param module an actual module deployed on the server
+     */
     private boolean matches(ModuleName name, ModuleInfo module) {
+        return matches(name, module, serverxmlmode);
+    }
+
+    private boolean matches(ModuleName name, ModuleInfo module, boolean overridenServerXMLMode) {
+        lazyInit();
         if (name.moduleName != null && !name.moduleName.equals(module.getName())) {
             return false;
         }
```

Comment:
Should this method be static? Does it need to `lazyInit`?

Reply:
Its a private method that is only called from methods that already lazyInit, so we could shave a nanosecond or two by removing that